compiler/rustc_passes/src/check_attr.rs

```diff
@@ -804,6 +804,37 @@ impl CheckAttrVisitor<'_> {
         true
     }
 
+    fn check_doc_tuple_variadic(&self, meta: &NestedMetaItem, hir_id: HirId) -> bool {
+        match self.tcx.hir().find(hir_id).and_then(|node| match node {
+            hir::Node::Item(item) => Some(&item.kind),
+            _ => None,
+        }) {
+            Some(ItemKind::Impl(ref i)) => {
+                if !matches!(&i.self_ty.kind, hir::TyKind::Tup([_])) {
+                    self.tcx
+                        .sess
+                        .struct_span_err(
+                            meta.span(),
+                            "`#[doc(tuple_variadic)]` can only be used on unary tuples",
```

Comment:
Maybe I'm alone on this but I found this error confusing, since tuple_variadic is intended specifically for implementations that cover tuples with arity greater than one.

How about: `#[doc(tuple_variadic)] can only be used on the first of a set of tuple trait implementations with varying arity` or similar?